pandas/core/frame.py

```diff
@@ -5070,14 +5070,38 @@ def notna(self) -> DataFrame:
     def notnull(self) -> DataFrame:
         return ~self.isna()
 
+    @overload
+    # https://github.com/python/mypy/issues/6580
```

Comment:
I'm pretty sure this comment has to go above the decorator?

Reply:
It doesn't in `pandas.core.frame.reset_index`, which doesn't seem to cause any problems. Also, it passed the typing portion of CI/Checks build, so I think we are fine. Happy to move it if there is a style preference though!

Reply:
I don't have a preference since it passes CI